nomad/fsm.go

```diff
@@ -2579,6 +2583,35 @@ func (s *nomadSnapshot) persistCSIVolumes(sink raft.SnapshotSink,
 	return nil
 }
 
+func (s *nomadSnapshot) persistServiceRegistrations(sink raft.SnapshotSink,
+	encoder *codec.Encoder) error {
+
+	// Get all the service registrations.
+	ws := memdb.NewWatchSet()
+	serviceRegs, err := s.snap.GetServiceRegistrations(ws)
+	if err != nil {
+		return err
+	}
+
+	for {
+		// Get the next item
+		raw := serviceRegs.Next()
+		if raw == nil {
+			break
+		}
```

Comment:
There's a nifty one-liner to iterate over the a go-memdb iterator if you prefer:

```go
for raw := serviceRegs.Next(); raw != nil; raw = serviceRegs.Next() {
  // ...
}
```